File: AutomaticBrewingCoffeeFE/interfaces/kiosk.ts
import { EBaseStatus } from "@/enum/base";
import { Device, DeviceIngredientHistory, DeviceModel } from "./device"
import { Store } from "./store";
import { Webhook } from "./webhook";

export interface Kiosk {
    kioskId: string;
    kioskVersionId: string;
    devices: Device[];
    location: string;
    status: EBaseStatus;
    installedDate: string;
    createdDate: string;
    updatedDate: string | null;
    kioskVersion: KioskVersion;
    position: string;
    warrantyTime: string;
    storeId: string;
    store: Store;
    kioskDevices: KioskDevice[];
    apiKey: string;
    webhooks?: Webhook[];
    menuId?: string;
    isOnline: boolean;
    isBusy: boolean;
}

export interface KioskDevice {
    device: Device;
    deviceId: string;
    kioskDeviceMappingId: string;
    kioskId: string;
    status: EBaseStatus
    isDisposed: boolean;
    deviceIngredientHistories: DeviceIngredientHistory[];
}

export interface KioskType {
    name: string;
    description: string;
    status: EBaseStatus;
    kioskTypeId: string;
    createdDate: string;
    updatedDate: string | null;
}

export interface KioskVersion {
    kioskTypeId: string;
    description: string;
    kioskVersionId: string;
    versionTitle: string;
    versionNumber: string;
    status: EBaseStatus;
    kiosks: Kiosk[];
    kioskVersionDeviceModelMappings: KioskVersionDeviceModelMappings[];
    kioskType: KioskType;
    createdDate: string;
    updatedDate: string | null;
}

export interface KioskVersionDeviceModelMappings {
    kioskVersionId: string;
    deviceModelId: string;
    kioskVersion: KioskVersion;
    deviceModel: DeviceModel;
    quantity: number;
}

File: AutomaticBrewingCoffeeBE/AutomaticBrewingCoffee.Main/Domains/Models/Kiosk.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AutomaticBrewingCoffee.Domain.Models;

public class Kiosk : BaseModel
{
    [Key] [StringLength(50)] [Required] public string KioskId { get; set; } = null!;

    [StringLength(50)] public string? KioskVersionId { get; set; }
    
    [StringLength(50)] public string? MenuId { get; set; }

    [StringLength(250)] public string? ApiKey { get; set; }

    [StringLength(250)] public bool IsRevoke { get; set; } = false;

    [StringLength(450)] public string? Hostname { get; set; }

    [StringLength(450)] public string? OriginServer { get; set; }

    [ForeignKey(nameof(KioskVersionId))] public virtual KioskVersion? KioskVersion { get; set; }

    [StringLength(450)] public string? Position { get; set; } = null!;

    public DateTime? WarrantyTime { get; set; }

    [StringLength(50)] [Required] public string StoreId { get; set; } = null!;

    [ForeignKey(nameof(StoreId))] public virtual Store? Store { get; set; } = null!;

    [StringLength(450)] [Required] public string Location { get; set; } = null!;

    [StringLength(10)] [Required] public string Status { get; set; } = null!;

    public DateTime InstalledDate { get; set; }

    public virtual ICollection<KioskDeviceMapping> KioskDevices { get; set; } = new List<KioskDeviceMapping>();
}